src/api.jl

```diff
@@ -0,0 +1,24 @@
+module api
+
+paths = ByteString[]
+const libvulkan = Libdl.find_library(["libvulkan", "vulkan", "libvulkan.so.1"], paths)
+@assert libvulkan != ""
+
+function _vk_func(func, ret_type, arg_types)
+  local args_in = Symbol[symbol("arg$i::$T")
+  for (i, T) in enumerate(arg_types.args)]
+  esc(quote
+    function $func($(args_in...))
+      ccall(($(string(func)), libvulkan),
+      $ret_type,
+      $arg_types,
+      $(args_in...))
+    end
+  end)
+end
```

Comment:
IIUC, this will have the same problem with https://github.com/ihnorton/Clang.jl/issues/152